sycl/include/sycl/ext/oneapi/annotated_arg/properties.hpp

```diff
@@ -327,15 +327,60 @@ struct is_valid_property<T, conduit_key::value_t> : std::true_type {};
 template <typename T>
 struct is_valid_property<T, stable_key::value_t> : std::true_type {};
 
+// Check if a given property is supported for annotated_arg
+// e.g. `alignment` is not supported for annotated_arg
+template <typename PropT>
+using is_property_value_of_annotated_arg =
+    is_property_value_of<PropT,
+                         annotated_arg<void, detail::empty_properties_t>>;
+
+// Check if a given property is supported for annotated_ptr
+template <typename PropT>
+using is_property_value_of_annotated_ptr =
+    is_property_value_of<PropT,
+                         annotated_ptr<void, detail::empty_properties_t>>;
+
+// Validate anntoated_arg/annotated_ptr
+template <typename T> struct validate_annotated_type : std::false_type {};
+
+// Partial specializations for validating annotated_arg
 template <typename T, typename... Props>
-struct check_property_list : std::true_type {};
+struct validate_annotated_type<annotated_arg<T, detail::properties_t<Props...>>>
+    : std::true_type {};
```

Comment:
Isn't the `detail::properties_t<>` be the only case that hits this? If so, we might as well make that clear in the definition:

```c++
template <typename T>
struct validate_annotated_type<annotated_arg<T, detail::properties_t<>>>
    : std::true_type {};
```